internal/mapset/mapset_test.go

```diff
@@ -0,0 +1,234 @@
+package mapset
+
+import (
+	"encoding/json"
+	"slices"
+	"testing"
+
+	"github.com/cedar-policy/cedar-go/internal/testutil"
+)
+
+func hashSetMustNotContain[T comparable](t *testing.T, s *MapSet[T], item T) {
+	testutil.FatalIf(t, s.Contains(item), "set %v unexpectedly contained item %v", s, 1)
+}
+
+func TestHashSet(t *testing.T) {
+	t.Run("empty set contains nothing", func(t *testing.T) {
+		s := Make[int]()
+		testutil.Equals(t, s.Len(), 0)
+		hashSetMustNotContain(t, s, 1)
+
+		s = Make[int](10)
+		testutil.Equals(t, s.Len(), 0)
+		hashSetMustNotContain(t, s, 1)
```

Comment:
Maybe checking for contains `0` would be more interesting, since it's the zero value.